### DISCUSSION TITLE
LaTeX

### DISCUSSION THREAD

**Discussion by mhx-gh** [newville/wxmplot#48] [Q&A] [answered]
I see little mentioned about text/labels.

Is there a way to have LaTeX label and annotate the plots?

-marcel

**Comment by newville** [comment, marked as answer]
@mhx-gh Yep, this is inherited exactly from matplotlib, which supports a very good subset of latex.  Just put latex math formatting between '$', and it should all work.  Several of the examples use latex for labels.   

Note that some escape sequences (say, `\t`) would be interpreted as python characters, so it is generally safest to use `r`-strings, so that labels might best be done with something like `r'$\phi(\tau)$ (degrees)' `

**Reply by mhx-gh** [reply]
Thanks! It was not clear from reading the info at Github (except for the source code :--)

**Reply by newville** [reply]
Of those 4 examples doing XY plots with labels in https://newville.github.io/wxmplot/examples.html, 2 do use latex, including the 1st one shown.  If you have suggestions for improving the docs, please make a pull request.

**Reply by mhx-gh** [reply]
I was looking at https://github.com/newville/wxmplot . My mistake...

**Comment by mhx-gh** [comment]
> demo.py will show several 2D Line plot examples, including a plot which uses a 
> timer to simulate a dynamic plot, updating the plot as fast as it can - typically 
> 10 to 30 times per second, depending on your machine.

That was my next question: would wxmplot be fast enough to incrementally plot the data from a (SPICE) circuitsimulator working in parallel. Typical data volume is 100MB, could be 8GB, so smart data handling would be needed. (I ask because Revolution EDA says it is using wxmplot).

**Comment by newville** [comment]
@mhx-gh I would suggest trying it.  The demo and stripchart examples are pretty simple.  On newish machines, the demo giving 30 updates per second seems pretty common, though that will depend some on gpu and cpu. 

As it turns out, image display (as in the example `image_scroll.py`) of 1000x1000 ndarrays as images can easily run at 15 fps.  

For sure, naive use of matplotlib code can be slow.   With some care, it is not too hard to make it be "fast enough" for displaying data.
